### PR TITLE
Add implementation for JiT Denoiser

### PR DESCRIPTION
**Pull request by nilsleh**
It has been fun exploring all the capabilities of Azula and trying to understand all the different approaches to implementing Score-based diffusion models. With the recent [JiT Paper](https://arxiv.org/abs/2511.13720) I thought it would be good to test my understanding and thinking about how this could fit into Azula. I am not sure if this is of interest, but I read the contribution guide and thought I would give it a shot, and get feedback.

### CODE REVIEW

File: azula/denoise.py [hunk 1]
```diff
@@ -266,3 +267,84 @@ def loss(self, x: Tensor, t: Tensor, **kwargs) -> Tensor:
...
+        r"""
+        Arguments:
+            x_t: A noisy tensor :math:`x_t`, with shape :math:`(B, *)`.
+            t: The time :math:`t`, with shape :math:`()` or :math:`(B)`.
+            kwargs: Optional keyword arguments.
+
+        Returns:
+            DiracPosterior where mean is the predicted x.
+        """
+        output = self.backbone(x_t, t, **kwargs)
```
[the remark this answers is not in the document]
Comment (francois-rozet) [reply azula/denoise.py]: Let $a = \alpha * x + \sigma \epsilon$ and $b = x / \alpha$. Then $E[x | a] = E[x | b]$. In other words, a denoiser should return the same output for two inputs that have the same signal-to-noise ratio.

To make this possible, the input of the backbone can be rescaled by $\alpha_t$. But this leads to very large inputs when t is large, for which NNs are not appropriate. Instead it is common to rescale by the input's std $\sqrt{\alpha_t^2 + \sigma_t^2}$. Then, regardless of the schedule, the input of the backbone is the same for the same signal-to-noise ratio. You could also rescale by $\alpha_t + \sigma_t$ which would correspond to the OT flow parameterization.

Again, if you don't do this, the backbone is tied to a single schedule, meaning that you have to use the same schedule during training and inference.

File: azula/denoise.py [hunk 2]
```diff
@@ -266,3 +267,84 @@ def loss(self, x: Tensor, t: Tensor, **kwargs) -> Tensor:
...
+        e = torch.randn_like(x) * self.noise_scale
+        z = alpha * x + sigma * e
+
+        v_target = (x - z) / sigma.clamp_min(self.t_eps)
+
+        x_pred = self(z, t, **kwargs).mean
+
+        v_pred = (x_pred - z) / sigma.clamp_min(self.t_eps)
+
+        loss = (v_target - v_pred) ** 2
```
[the remark this answers is not in the document]
Comment (francois-rozet) [reply azula/denoise.py]: Let's say instead of $\alpha_t = 1-t$ and $\sigma_t = t$ the schedule is $\alpha_t = 1 - t^n$ and $\sigma_t = t^n$, do you think the loss weight should be different?
